src/Jackett.Common/Indexers/FunFile.cs

```diff
@@ -126,14 +116,17 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                     var qTimeAgo = row.QuerySelector("td:nth-of-type(6)");
                     var qSize = row.QuerySelector("td:nth-of-type(8)");
 
-                    var catStr = qCatLink.GetAttribute("href").Split('=')[1].Split('&')[0];
-                    release.Category = MapTrackerCatToNewznab(catStr);
+                    if (qDownloadLink == null)
+                        throw new Exception("Download link not found. Make sure you can download from the website.");
```

Comment:
If the only time a download link is expecting to be missing is when they're disabled for every torrent in the tracker, then this is ok, but the exception should reflect this as below. If only some of the results don't have a download link for whatever reason (like 720pier is a forum with most, not all, posts including torrent downloads as example), then we should be using `continue` instead.
```suggestion
                    if (qDownloadLink == null)
                        throw new Exception("Download links not found. Make sure you can download from the website.");
```

Reply:
In this case the user will see all download links or none.